qadence/ml_tools/optimize_step.py

```diff
@@ -6,12 +6,15 @@
 from torch.nn import Module
 from torch.optim import Optimizer
 
+from qadence.ml_tools.data import data_to_device
+
 
 def optimize_step(
     model: Module,
     optimizer: Optimizer,
     loss_fn: Callable,
     xs: dict | list | torch.Tensor | None,
+    device: torch.device = torch.device("cpu"),
```

Comment:
what about making device default to None? this way if everything already is in it's place, it won't be moved. And we only move if explictly requested.

```
>>> t = torch.randn(2,2)
>>> t
tensor([[-0.2409, -0.3257],
        [-0.0846,  1.2733]])
>>> t.device
device(type='cpu')
>>> t.dtype
torch.float32
>>> t=t.to(torch.device("cuda:0"), dtype=torch.float64)
>>> t
tensor([[-0.2409, -0.3257],
        [-0.0846,  1.2733]], device='cuda:0', dtype=torch.float64)
>>> t.to(None)
tensor([[-0.2409, -0.3257],
        [-0.0846,  1.2733]], device='cuda:0', dtype=torch.float64)
```

Reply:
good idea, added